### PR TITLE
many: refactor in preparation for 'snap start'

### PR DESCRIPTION
**Pull request by chipaca**
wrappers: move {Start,Stop}SnapServices(*snap.Info) to {Start,Stop}Services([]*snap.AppInfo)
snap/info: gave *snap.Info a Services() method which returns a []*snap.AppInfo of services
overlord/snapstate: propagate the refactor up to (but not including) the task level

### CODE REVIEW

File: snap/info.go [hunk 1]
```diff
@@ -280,6 +280,19 @@ func (s *Info) NeedsClassic() bool {
 	return s.Confinement == ClassicConfinement
 }
 
+// Services returns a list of the apps that have "daemon" set.
+func (s *Info) Services() []*AppInfo {
```
Comment (pedronis) [inline snap/info.go]: test in info_test.go ?

File: snap/info.go [hunk 2]
```diff
@@ -482,6 +495,11 @@ func (app *AppInfo) Env() []string {
 	return env
 }
 
+// IsService returns whether app represents a daemon/service.
+func (app *AppInfo) IsService() bool {
```
Comment (pedronis) [inline snap/info.go]: test?

File: overlord/snapstate/handlers.go [hunk 3]
```diff
@@ -690,10 +690,15 @@ func (m *SnapManager) startSnapServices(t *state.Task, _ *tomb.Tomb) error {
 	if err != nil {
 		return err
 	}
+	svcs := currentInfo.Services()
+	// TODO:
```
Comment (pedronis) [inline overlord/snapstate/handlers.go]: I fear is better to bite the bullet here given that you need to touch all the tests anyway.

File: overlord/snapstate/backend_test.go [hunk 4]
```diff
@@ -397,18 +397,16 @@ func (f *fakeSnappyBackend) LinkSnap(info *snap.Info) error {
 	return nil
 }
 
-func (f *fakeSnappyBackend) StartSnapServices(info *snap.Info, meter progress.Meter) error {
+func (f *fakeSnappyBackend) StartServices(svcs []*snap.AppInfo, meter progress.Meter) error {
 	f.ops = append(f.ops, fakeOp{
-		op:   "start-snap-services",
-		name: info.MountDir(),
+		op: "start-snap-services",
```
Comment (pedronis) [inline overlord/snapstate/backend_test.go]: this is a bit "cheating", if one now replaces all the .Services() with nil tests would still pass. We probably want a couple of tests with services in snapstate now and then you can have the ifs, notice that the way to achieve that is to have a name like service-snap handled in backend_test.go in fake ReadInfo

Comment (pedronis) [inline overlord/snapstate/backend_test.go]: see alias-snap there for example

Comment (pedronis) [inline overlord/snapstate/backend_test.go]: now you can keep name:  or something like that,  svcs[0].Snap.MountDir()

File: snap/info.go (hunk 1 above)
Comment (chipaca) [inline snap/info.go]: done

File: snap/info.go (hunk 2 above)
Comment (chipaca) [inline snap/info.go]: done

**Pull request merged**